kogito-workitems/kogito-rest-workitem/src/main/java/org/kogito/workitem/rest/RestWorkItemHandler.java

```diff
@@ -139,6 +140,10 @@ public void executeWorkItem(KogitoWorkItem workItem, KogitoWorkItemManager manag
         authDecorators.forEach(d -> d.decorate(workItem, parameters, request));
         paramsDecorator.decorate(workItem, parameters, request);
         HttpResponse<Buffer> response = method.equals(HttpMethod.POST) || method.equals(HttpMethod.PUT) ? request.sendJsonAndAwait(bodyBuilder.apply(parameters)) : request.sendAndAwait();
+        int statusCode = response.statusCode();
+        if (statusCode >= 300) {
```

Comment:
```suggestion
        if (statusCode >= 400) {
```

`300` codes are for redirection. We should either log or have the internal factory do the redirect. Not sure if we want an exception here?